New resources `azurerm_mssql_virtual_machine_group` and `azurerm_mssql_virtual_machine_availability_group_listener` (#22808)
hashicorp/terraform-provider-azurerm: website/docs/r/mssql_virtual_machine_availability_group_listener.html.markdown

  sql_virtual_machine_group_id = azurerm_mssql_virtual_machine_group.example.id

  load_balancer_configuration {
    load_balancer_id = data.azurerm_lb.example.id
    probe_port       = 51572

    private_ip_address {
      ip_address = "10.0.2.11"
      subnet_id  = data.azurerm_subnet.example.id
    }

    sql_virtual_machine_ids = [
      azurerm_mssql_virtual_machine.example[0].id,
      azurerm_mssql_virtual_machine.example[1].id
    ]
  }

  replica {
    sql_virtual_machine_id = azurerm_mssql_virtual_machine.example[0].id
    role                   = "Primary"
    commit                 = "Synchronous_Commit"
    failover               = "Automatic"
    readable_secondary     = "All"
  }

  replica {
    sql_virtual_machine_id = azurerm_mssql_virtual_machine.example[1].id
    role                   = "Secondary"
    commit                 = "Asynchronous_Commit"
    failover               = "Manual"
    readable_secondary     = "No"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for the Microsoft SQL Virtual Machine Availability Group Listener. Changing this forces a new resource to be created.

> * `sql_virtual_machine_group_id` - (Required) ID of the SQL Virtual Machine Group to create the listener. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `sql_virtual_machine_group_id` - (Required) The ID of the SQL Virtual Machine Group for which to create the listener. Changing this forces a new resource to be created.
```

* `availability_group_name` - (Optional) Name of the Availability Group. Changing this forces a new resource to be created.

* `load_balancer_configuration` - (Optional) A `load_balancer_configuration` block as defined below. Changing this forces a new resource to be created.

~> **NOTE:** Either one of `load_balancer_configuration` or `multi_subnet_ip_configuration` must be specified.

* `multi_subnet_ip_configuration` - (Optional) One or more `multi_subnet_ip_configuration` blocks as defined below. Changing this forces a new resource to be created.

* `port` - (Optional) Port of the listener. Changing this forces a new resource to be created.

* `replica` - (Optional) One or more `replica` blocks as defined below. Changing this forces a new resource to be created.

---

A `load_balancer_configuration` block supports the following:

* `load_balancer_id` - (Required) The ID of the Load Balancer. Changing this forces a new resource to be created.

* `private_ip_address` - (Required) A `private_ip_address` block as defined below. Changing this forces a new resource to be created.

> * `probe_port` - (Required) Probe port of the listener. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `probe_port` - (Required) The probe port of the listener. Changing this forces a new resource to be created.
```

> * `sql_virtual_machine_ids` - (Required) Specifies a list of IDs of the SQL Virtual Machine. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `sql_virtual_machine_ids` - (Required) Specifies a list of SQL Virtual Machine IDs. Changing this forces a new resource to be created.
```

~> **NOTE:** `sql_virtual_machine_ids` should match with the SQL Virtual Machines specified in `replica`.

---

A `multi_subnet_ip_configuration` block supports the following:

* `private_ip_address` - (Required) A `private_ip_address` block as defined below. Changing this forces a new resource to be created.

> * `sql_virtual_machine_id` - (Required) ID of the Sql Virtual Machine. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `sql_virtual_machine_id` - (Required) The ID of the Sql Virtual Machine. Changing this forces a new resource to be created.
```

~> **NOTE:** `sql_virtual_machine_id` should match with the SQL Virtual Machines specified in `replica`.

---

A `private_ip_address` block supports the following:

* `ip_address` - (Required) IP Address. Changing this forces a new resource to be created.

* `subnet_id` - (Required) The ID of the Subnet. Changing this forces a new resource to be created.

---

A `replica` block supports the following:

> * `commit` - (Required) Replica commit mode in availability group. Possible values are `Synchronous_Commit` and `Asynchronous_Commit`. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `commit` - (Required) The replica commit mode in for the availability group. Possible values are `Synchronous_Commit` and `Asynchronous_Commit`. Changing this forces a new resource to be created.
```

> * `commit` - (Required) T replica commit mode for the availability group. Possible values are `Synchronous_Commit` and `Asynchronous_Commit`. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `commit` - (Required) The replica commit mode for the availability group. Possible values are `Synchronous_Commit` and `Asynchronous_Commit`. Changing this forces a new resource to be created.
```

> * `failover` - (Required) Replica failover mode in availability group. Possible values are `Manual` and `Automatic`. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `failover` - (Required) The replica failover mode for the availability group. Possible values are `Manual` and `Automatic`. Changing this forces a new resource to be created.
```

> * `readable_secondary` - (Required) Replica readable secondary mode in availability group. Possible values are `No`, `Read_Only` and `All`. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `readable_secondary` - (Required) The replica readable secondary mode for the availability group. Possible values are `No`, `Read_Only` and `All`. Changing this forces a new resource to be created.
```

> * `role` - (Required) Replica Role in availability group. Possible values are `Primary` and `Secondary`. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `role` - (Required) The replica role for the availability group. Possible values are `Primary` and `Secondary`. Changing this forces a new resource to be created.
```
